### ISSUE TITLE
Can't use relation fields in title_format

### ISSUE THREAD

**Issue by philbates35** [bolt/bolt#3849]
I would like to be able to use a field from one of my content type's relations in it's admin title (set with the `title_format` config).

Use case: I have a `repeating_events` content type, and a `repeating_event_instances` content type. Each `repeating_event` is linked to many `repeating_event_instances`. Each `repeating_event_instance` has a single `date` field, and it is linked to a single `repeating_event`.

I would like the admin screen to display like the following example:

```
01/01/2015 - Repeating Event 1
02/01/2015 - Repeating Event 2
08/01/2015 - Repeating Event 1
```

I've had a quick look in the source, and it seems that it `title_format` you can only pass the field names of the fields of the content type in question. Meaning that it is currently only possible for my admin screen to look like this:

```
01/01/2015
02/01/2015
08/01/2015
```

I'm sure you can see how this would be confusing to the user.

I envisage the updated `title_format` to allow you to do something like this:

```
repeating_event_instances:
    name: Repeating Event Instances
    singular_name: Repeating Event Instance
    title_format: [ date, relations.repeating_events.name ]
    fields:
        date:
            type: date
            required: true
    relations:
        repeating_events:
            multiple: false
            label: Select a repeating event
            order: name
```

Let me know what you think!

Thanks.

**Comment by Pinpickle** [comment]
May I suggest writing a [custom content class](https://docs.bolt.cm/extensions/essentials#overriding-default-8216content8217-class)? I have done something very similar with title logic, it shouldn't be too hard to accomplish what you want (this is not tested)

```php
class RepeatingEventInstance extends \Bolt\Content {
    public function getTitle()
    {
        $relatedEvents = $this->related('repeating_events');
        $relatedEvent = $relatedEvents[0];
        
        return $this->values['date'] . ' - ' . ($releatedEvent ? $relatedEvent->values['name'] : 'Unknown Event');
    }
}
```

And in your `contenttypes.yml`

```yml
repeating_event_instances:
    # All of your stuff
    class: \Your\Namespace\RepeatingEventInstance
```

**Comment by philbates35** [comment]
Seems sensible, I'll give it a go tonight thanks but I'm sure it will work. I guess it would still be nice to be able to do it directly from within the `title_format`(a future nice-to-have maybe?), but this workaround will suffice for now. Thanks.

**Comment by Intendit** [comment]
I suspect that this is an issue for all "non-normal" content. I wanted to show the name for the template from a templateselect as part of the title, but can only show the filename, not the name I've given it in my theme's config.yml. Of course one can always use the custom content classes, but that might not be the ideal long term solution.

**Comment by philbates35** [comment]
So I've just tried to implement the suggestion in post 2, and it almost worked, but not quite.

In the main listing of `repeating_events` on `/bolt/overview/repeating-event-instances` it works fine. However, it doesn't work in the "Last modified Repeating Event Instances" pane on the right hand side, and it also doesn't work in the hover menu in the left hand pane (i.e when I hover over Repeating Event Instances management button), both of which show, for example "01/01/2015 - Unknown Event".

Using `var_dump($this->relation)` at the start of the overridden `getTitle()` method shows that `$this->relation` is `null` in the right and left hand panes, but is correctly populated with the relation in the main content. Any ideas?

Here is my full custom content class:

```
<?php

namespace Acme;

use Bolt\Content;

class RepeatingEventInstanceContent extends Content
{
    /**
     * Get the title, name, caption or subject.
     *
     * @return string
     */
    public function getTitle()
    {
        $repeatingEvents = $this->related('repeating-events');
        $repeatingEvent = $repeatingEvents[0];

        return $this->values['date'] . ' - ' . ($repeatingEvent ? $repeatingEvent->values['name'] : 'Unknown Event');
    }
}
```

**Comment by SahAssar** [comment]
To me it seems like the solutions posted solve the edge cases where it would be used, and that limiting usage of title_format might be a good thing (even though Intendit is my work github account). Custom content classes should solve this perfectly fine if the issues that @philbates35 mentioned are fixed (if they aren't already in 2.3).

**Comment by SahAssar** [comment]
Closing this per my comment from Oct 26, 2015, if you feel this is still an issue that needs resolving in core please feel free to reopen :)
